Src/Web/Web.Net45/Implementation/HttpRequestExtensions.cs

```diff
@@ -18,7 +18,7 @@ public static HttpCookie UnvalidatedGetCookie(this HttpRequest httpRequest, stri
         public static string UnvalidatedGetHeader(this HttpRequest httpRequest, string headerName)
         {
             string value = httpRequest.Unvalidated.Headers[headerName];
```

Comment:
we should always expect a header value to be null

EnforceMaxLength has Debug.Assert that value is not null preventing to run tests locally in debug.

Instead, I can remove Debug.Assert from the `EnforceMaxLength`, however I don;t know what was the motivation putting it there in the first place.

@MS-TimothyMothra,
It seems you've done this change. could you please advise what is the best approach here?

Reply:
If it's causing problems, please remove it.
I added it because it seemed like it was the convention.
I have a backlog item to remove it myself and create actual unit tests for that Common project.

Reply:
will do.

I also wonder what is the need for enforcing max length on the incoming side? If request made it to the service, why would we trim it? Where 1024 came from?

Reply:
This risk was identified from SDL review.
Request headers are tracked in a dictionary and an attacker could spam an application with very large header values and force an out of memory exception.
Sergey gave me the 1024 value. That should be exaggeratedly larger than we need, but still small enough to protect ourselves from an attack.